src/client/FhRef.h

```diff
@@ -0,0 +1,12 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#ifndef CEPH_CLIENT_FHREF_H
+#define CEPH_CLIENT_FHREF_H
+
+#include <boost/intrusive_ptr.hpp>
+class Fh;
+void intrusive_ptr_add_ref(Fh *fh);
+void intrusive_ptr_release(Fh *fh);
+typedef boost::intrusive_ptr<Fh> FhRef;
```

Comment:
> Hmm...just today, @batrick gave a talk on new pointer types, shared_ptr in particular. Should we be expanding the use of intrusive pointers here? Does this give us something that the new std:: pointers don't?
> 

I will check it to see which ones could switch to the shared ptr, but for the InodeRef/CapRef couldn't.

> It would be nice too if there were more explanation of the changes in the changelogs. 

I will add more detailed explaination about it.

>This patch series overall just jumps into making a lot of these changes without explaining why and what the benefit of these changes are. I can imagine that we do need a FhRef object, but it would be nice to understand why it's necessary and how this relates to the BCL (Big Client Lock).

Okay, I will add it.